castle/cms/tiles/calendar.py

```diff
@@ -23,10 +24,60 @@ def format_date(dt):
             pass
 
 
+class ICalendarTileSchema(Interface):
+
+    form.widget(query=QueryFieldWidget)
+    query = schema.List(
+        title=u'Base query',
+        description=u"This query can be customized base on user selection",
+        value_type=schema.Dict(value_type=schema.Field(),
+                               key_type=schema.TextLine()),
+        required=False,
+        default=[{
+            u'i': u'portal_type',
+            u'v': [u'Event'],
+            u'o': u'plone.app.querystring.operation.selection.any'}]
+    )
+
+    sort_on = schema.TextLine(
+        title=u'Sort on',
+        description=u"Sort on this index",
+        required=False,
+    )
+
+    sort_reversed = schema.Bool(
+        title=u'Reversed order',
+        description=u'Sort the results in reversed order',
+        required=False,
+        default=False,
+    )
+
+    limit = schema.Int(
+        title=u'Limit',
+        description=u'Limit Search Results',
+        required=False,
+        default=250,
+        min=1,
+    )
+
+
 class CalendarTile(BaseTile):
+    implements(ICalendarTileSchema)
 
     def get_query(self):
         parsed = parse_query_from_data(self.data, self.context)
+        # If parsed is empty and the self.data doesn't contain any of the variables below
+        # then we have a new calendar to set the defaults on.
+        if parsed == {}:
+            if self.data.get('query', None) is None:
+                if self.data.get('limit', None) is None:
+                    if self.data.get('sort_reversed', None) is None:
+                        if self.data.get('sort_on', None) is None:
+                            self.data[u'limit'] = ICalendarTileSchema["limit"].default
+                            self.data[u'query'] = ICalendarTileSchema["query"].default
+                            self.data[u'sort_reversed'] = ICalendarTileSchema["sort_reversed"].default
+                            return self.get_query()
+
         parsed['sort_order'] = 'reverse'
```

Comment:
From what I can see so far, (in master as well) this line here is the only one that there might be a reason to change from master, because it is setting the sort_order to 'reverse' regardless of the value of the sort_reversed field in the schema.  Since a calendar is a time-ordered thing I'm not exactly sure what the sorting here is doing but we can look into it if needed.

Reply:
I agreed, removed it.